releasenotes/notes/plot-hist-797bfaeea2156c53.yaml

```
@@ -0,0 +1,11 @@
+---
+features:
+  - |
+    Adds ``plot_distribution``for plotting distributions over quasi-probabilities.
+    This is suitable for ``Counts``, ``QuasiDistribution`` and ``ProbDistribution``.
+    Raw `dict` can be passed as well.
+upgrade:
+  - |
+    Modifies ``plot_histogram`` to return an actual histogram of discrete binned values.
+    As the values are discrete, only ``Counts`` and ``dict`` objects are allowed.
+    ``dict`` objects are not validated for correctness, e.g. discrete values.
```

We probably should be a bit more explict on what exactly changed, and how to
adapt for it someone really cares.

```suggestion
    The :func:`~.plot_histogram` function has been modified to return an actual
    histogram of discrete binned values. The previous behavior for the function
    was despite the name to actually generate a visualization of the distribution
    of the input. Due to this disparity between the name of the function and the behavior
    the function behavior was changed so it's actually generating a proper histogram
    of discrete data now. If you wish to preserve the previous behavior of plotting a
    probability distribution of the counts data you can leverage the :func:`~.plot_distribution` to generate an
    equivalent graph. For example, the previous behavior of
    ``plot_histogram({'00': 512, '11': 500})`` can be re-created with:
    
    .. jupyter-execute::
    
        from qiskit.visualization import plot_distribution
        import matplotlib.pyplot as plt
        
        ax = plt.subplot()
        plot_distribution({'00': 512, '11': 500}, ax=ax)
        ax.set_ylabel('Probabilities')
        ax.get_figure()
```